### PR TITLE
Added shop.java

### PR DESCRIPTION
**Pull request by ArtemSadovenko**

### CODE REVIEW

File: src/main/java/dto/ShopDTO.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import java.time.LocalDateTime;
+
+public class ShopDTO {
+    private Long id;
+    private String shopName;
+    private String address;
+    private long shopOwnerID;
+    private LocalDateTime timeOpen;
+    private LocalDateTime timeClose;
+    private long holidayID;
```
Comment (dmytro-krupa) [inline src/main/java/dto/ShopDTO.java]: Long

File: src/main/java/shippingService/entity/Shop.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@NoArgsConstructor
+@Entity
+@Table(name = "Shop")
+public class Shop {
+    @Id
+    @NonNull
+    private Long id;
+    private String shopName;
+    private String address;
+    private long shopOwnerID;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/entity/Shop.java]: @ One-To-One
private User shopOwner

File: src/main/java/shippingService/entity/Shop.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private LocalDateTime timeOpen;
+    private LocalDateTime timeClose;
+    private long holidayID;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/entity/Shop.java]: private List Holiday holidays

File: src/main/java/shippingService/entity/Shop.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@Table(name = "Shop")
+public class Shop {
+    @Id
+    @NonNull
+    private Long id;
+    private String shopName;
+    private String address;
+    private long shopOwnerID;
+    private LocalDateTime timeOpen;
+    private LocalDateTime timeClose;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/entity/Shop.java:44]: ALTER TABLE table_name
MODIFY COLUMN column_name datatype;

File: src/main/java/shippingService/dto/ShopDTO.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    private long id;
+    private String shopName;
+    private String address;
+    @Id
+    private long shopOwnerID;
+    private LocalDateTime timeOpen;
+    private LocalDateTime timeClose;
+    @Id
+    private long holidayID;
+}
```
Comment (RIO489) [inline src/main/java/shippingService/dto/ShopDTO.java:33]: remove @ Id from DTO

File: src/main/java/shippingService/service/impl/ShopServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    @Override
+    public void delete(long id) {
+        shopRepository.deleteById(id);
+    }
+
+    @Override
+    public ShopDTO update(ShopDTO shopDTO) {
+//        Shop shop = shopRepository.findById(shopDTO.getId()).get();
+        delete(shopDTO.getId());
+        create(shopDTO);
```
Comment (RIO489) [inline src/main/java/shippingService/service/impl/ShopServiceImpl.java]: instead use repo`s method: save(*);

File: src/main/java/shippingService/dto/ShopDTO.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    private String shopName;
+    private String address;
+//    @DateTimeFormat(style = "hh:mm")
+//    @JsonFormat(shape=JsonFormat.Shape.STRING, pattern="hh:mm")
+    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
+    private LocalDateTime timeOpen;
+    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
+    private LocalDateTime timeClose;
+    private List<Holiday> holidays;
+    private User shopOwner;
```
Comment (RIO489) [inline src/main/java/shippingService/dto/ShopDTO.java]: Instead use UserDTO

File: src/main/java/shippingService/service/HolidayService.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+import shippingService.dto.HolidayDTO;
+import shippingService.repository.HolidayRepository;
+
+import java.util.List;
+
+public interface HolidayService {
+    void create(HolidayDTO holidayDTO);
+
+    void update(HolidayDTO holidayDTO);
```
Comment (RIO489) [inline src/main/java/shippingService/service/HolidayService.java]: not void,use DTO instead for create

File: src/main/java/shippingService/service/impl/ShopServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    @Transactional
+    public ShopDTO update(ShopDTO shopDTO) {
+//        Shop shop = shopRepository.findById(shopDTO.getId()).get();
+        delete(shopDTO.getId());
+        create(shopDTO);
+
+        shopRepository.save(MapperShop.ToEntity(shopDTO));
+        //save
+        //todo
```
Comment (RIO489) [inline src/main/java/shippingService/service/impl/ShopServiceImpl.java]: Just leave save method and delete anything left,that all
something like that:
shopRepository.save(mapperShop.toEntity(dto))

**Review by RIO489 — APPROVED** [review]

**Pull request merged**